sources/decompiler.cpp

```diff
@@ -122,22 +116,21 @@ bool Decompiler::write_dat(const QString& output_dir) {
     file_content.append(header);
     file_content.append(functions);
 
-    QDir dir(output_dir);
-    if (!dir.exists()) dir.mkpath(".");
+    if (!fs::exists(output_dir)) fs::create_directories(output_dir);
 
-    QString output_path = output_dir + "\\" + QString::fromStdString(current_tf.getName()) + ".dat";
-    QFile file(output_path);
+    fs::path output_path = output_dir / (current_tf.getName() += ".dat");
```

Comment:
I wish we could use the simple `+` operator here, but it's not currently exposed by `std::filesystem::path`. Sadly  only `+=` is.

Reply:
lol, using that was a bug anyways, so this is irrelevant now. i found another way around it.